### PR TITLE
fix: get vmss name and resource group from vm ID if the provider ID o…

### PR DESCRIPTION
**Pull request by nilo19**
…f the node is empty

#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

When a new node is just joining the cluster, the providerID can be empty. In this case, we will skip the VMSS in ensureVMSSInPool. This PR fixes this by using the VM ID instead.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
fix: get vmss name and resource group from vm ID if the provider ID of the node is empty
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/provider/azure_vmss.go [hunk 1]
```diff
@@ -47,11 +47,13 @@ var (
...
+	scaleSetNameRE           = regexp.MustCompile(`.*/subscriptions/(?:.*)/Microsoft.Compute/virtualMachineScaleSets/(.+)/virtualMachines(?:.*)`)
+	resourceGroupRE          = regexp.MustCompile(`.*/subscriptions/(?:.*)/resourceGroups/(.+)/providers/Microsoft.Compute/virtualMachineScaleSets/(?:.*)/virtualMachines(?:.*)`)
+	vmssIPConfigurationRE    = regexp.MustCompile(`.*/subscriptions/(?:.*)/resourceGroups/(.+)/providers/Microsoft.Compute/virtualMachineScaleSets/(.+)/virtualMachines/(.+)/networkInterfaces(?:.*)`)
+	vmssPIPConfigurationRE   = regexp.MustCompile(`.*/subscriptions/(?:.*)/resourceGroups/(.+)/providers/Microsoft.Compute/virtualMachineScaleSets/(.+)/virtualMachines/(.+)/networkInterfaces/(.+)/ipConfigurations/(.+)/publicIPAddresses/(.+)`)
+	vmssVMResourceIDTemplate = `/subscriptions/(?:.*)/resourceGroups/(.+)/providers/Microsoft.Compute/virtualMachineScaleSets/(.+)/virtualMachines/(?:\d+)`
+	vmssVMResourceIDRE       = regexp.MustCompile(vmssVMResourceIDTemplate)
+	vmssVMProviderIDRE       = regexp.MustCompile(fmt.Sprintf("%s%s", "azure://", vmssVMResourceIDTemplate))
```
Comment (feiskyer) [inline pkg/provider/azure_vmss.go:56]: vmssVMResourceIDRE and vmssVMProviderIDRE could be merged together with `.*` prefix

**Comment by feiskyer** [comment]
The test features didn't seem related to this change

/retest

File: pkg/provider/azure_vmss.go (hunk 1 above)
Comment (nilo19) [reply pkg/provider/azure_vmss.go:56]: Yes, but I want to make sure the provider ID will start with the `azure://`.

**Pull request merged**